app/src/main/java/com/doublesp/coherence/activities/AddFriendActivity.java

```diff
@@ -1,58 +1,114 @@
+        userFriends = new ArrayList<>();
+        userFriends.add(ConstantsAndUtils.getOwner(this));
 
         final LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
         linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
         mRecyclerView.setLayoutManager(linearLayoutManager);
+        DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(this,
+                linearLayoutManager.getOrientation());
+        mRecyclerView.addItemDecoration(dividerItemDecoration);
 
         mFirebaseRecyclerAdapter = new FirebaseRecyclerAdapter<User, AddFriendViewHolder>(
                 User.class, R.layout.single_add_friend, AddFriendViewHolder.class,
                 mUsersDatabaseReference) {
 
             @Override
-            protected void populateViewHolder(AddFriendViewHolder holder, User user, int position) {
+            protected void populateViewHolder(final AddFriendViewHolder holder, User user,
+                    int position) {
+                String name = user.getName();
                 String email = user.getEmail();
-                if (userFriends.contains(email)) return;
-                holder.mEmail.setText(email.replace(",", "."));
+                String emailDecoded = email.replace(",", ".");
+                if (name == null || name.isEmpty()) name = emailDecoded;
+
+                ColorGenerator generator = ColorGenerator.MATERIAL;
+                int color = generator.getColor(emailDecoded);
+
+                TextDrawable.IBuilder initialConfig = TextDrawable.builder()
+                        .beginConfig()
+                        .endConfig()
+                        .roundRect(10);
+                TextDrawable initial = initialConfig.build(String.valueOf(name.charAt(0)), color);
+
+                holder.mInitials.setImageDrawable(initial);
+                holder.mEmail.setText(emailDecoded);
+                holder.mName.setText(user.getName());
+
+                if (email.equals(ConstantsAndUtils.getOwner(AddFriendActivity.this))) {
+                    holder.mAddFriendButton.setImageResource(R.drawable.ic_check);
+                    return;
+                }
+
+                mListsDatabaseReference.child(ConstantsAndUtils.getOwner(AddFriendActivity.this))
+                        .child(email).addListenerForSingleValueEvent(
+
+                        new ValueEventListener() {
+                            @Override
+                            public void onDataChange(DataSnapshot dataSnapshot) {
+                                User thisUser = dataSnapshot.getValue(User.class);
+                                if (thisUser != null) {
+                                    holder.mAddFriendButton.setImageResource(R.drawable.ic_check);
+                                } else {
+                                    holder.mAddFriendButton.setImageResource(
+                                            R.drawable.ic_add_green);
```

Comment:
actually you don't need new asset for these buttons, just use R.drawable.ic_create and set tint to green. but nvm it's not big deal either.